torchdata/datapipes/iter/util/prefetcher.py

```diff
@@ -93,7 +95,11 @@ def __iter__(self):
                 self.thread.start()
                 while prefetch_data.run_prefetcher:
                     if len(prefetch_data.prefetch_buffer) > 0:
-                        yield prefetch_data.prefetch_buffer.popleft()
+                        item = prefetch_data.prefetch_buffer.popleft()
+                        if isinstance(item, Exception):
+                            prefetch_data.run_prefetcher = False
+                            raise item
+                        yield item
```

Comment:
But keep this while loop working until `Exception` is popped from buffer.

It seems `run_prefetcher` needs to have three states:
- Normal
- Stop thread while loop
- Stop iter while loop